### DISCUSSION TITLE
SiteRouter.Refresh Exception thrown after logging in with external oidc provider

### DISCUSSION THREAD

**Discussion by Justincale** [oqtane/oqtane.framework#4900] [Q&A] [answered]
Hi, 

I am using an external oidc login provider which is working, although i have noticed Oqtane throws the following exception in SiteRouter.Refresh after the login completes: "Exception of type 'Microsoft.AspNetCore.Components.NavigationException' was thrown."

This occurs on the following line: NavigationManager.NavigateTo(_absoluteUri.Replace("?reload=post", "").Replace("&reload=post", ""), true);

Where the Uri is "http://localhost:44357/?reload=post"

the code continues and Oqtane seems fine, just not sure if this would be causing any other issues down the line.

Thanks

**Comment by leigh-pointer** [comment, marked as answer]
@Justincale This is a know issue with .net core. Have a look here. I am sure this will help you https://www.youtube.com/watch?v=1lsjpfdqBk0&t=2321s

**Reply by Justincale** [reply]
Holy cow, how have i not seen this video before. Part of integrating my system into Oqtane has been an upgrade to .net 9 and i have fallen pray to everything mentioned in this video. Thanks :)

**Comment by sbwalker** [comment]
As @leigh-pointer mentioned, this is not an issue. It only occurs in development when a debugger is attached to the process. You can safely ignore the exception in Visual Studio.
